bootstrap/updater.go

```diff
@@ -143,51 +145,57 @@ func (u *Updater) Load(ctx context.Context) error {
 		if err = u.updateAndNotify(ctx, verified); err != nil {
 			return err
 		}
-		u.logger.With().Info("loaded boostrap file", log.Inline(verified))
+		u.logger.With().Info("loaded bootstrap file", log.Inline(verified))
 		u.addUpdate(verified.Data.Epoch, verified.Persisted[len(verified.Persisted)-suffixLen:])
 	}
 	return nil
 }
 
-func (u *Updater) Start(ctx context.Context) error {
+func (u *Updater) Start() error {
 	if len(u.cfg.DataDir) == 0 {
 		return fmt.Errorf("data dir not set %s", u.cfg.DataDir)
 	}
 	u.once.Do(func() {
 		u.eg.Go(func() error {
+			ctx := log.WithNewSessionID(context.Background())
 			if err := u.Load(ctx); err != nil {
 				return err
 			}
-			wait := time.Duration(0)
 			u.logger.With().Info("start listening to update",
 				log.String("source", u.cfg.URL),
 				log.Duration("interval", u.cfg.Interval),
 			)
 			for {
+				if err := u.DoIt(ctx); err != nil {
+					updateFailureCount.Add(1)
+					u.logger.With().Debug("failed to get bootstrap update", log.Err(err))
+				}
 				select {
-				case <-ctx.Done():
-					return ctx.Err()
-				case <-time.After(wait):
-					ctx := log.WithNewSessionID(ctx)
-					if err := u.DoIt(ctx); err != nil {
-						updateFailureCount.Add(1)
-						u.logger.With().Debug("failed to get bootstrap update", log.Err(err))
-					}
+				case <-u.stop:
+					return nil
+				case <-time.After(u.cfg.Interval):
 				}
-				wait = u.cfg.Interval
 			}
 		})
 	})
 	return nil
 }
 
-func (u *Updater) Close() {
+func (u *Updater) Close() error {
+	select {
+	case <-u.stop: // prevent calling Close twice from a panic
```

Comment:
i like this protection, but not sure about the comment. from a panic? how?

Reply:
Yeah, don't know what I was thinking when I wrote that comment... Updated it.